test/integration/Cardano/NetworkLayer/HttpBridgeSpec.hs

```diff
@@ -0,0 +1,102 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Cardano.NetworkLayer.HttpBridgeSpec
+    ( spec
+    ) where
+
+import Prelude
+
+import Cardano.Launcher
+    ( Command (..), launch )
+import Cardano.NetworkLayer
+    ( NetworkLayer (..) )
+import Cardano.NetworkLayer.HttpBridge
+    ( HttpBridgeError (..) )
+import Cardano.Wallet.Primitive
+    ( Block (..), BlockHeader (..), Hash (..), SlotId (..) )
+import Control.Concurrent
+    ( threadDelay )
+import Control.Concurrent.Async
+    ( async, cancel )
+import Control.Exception.Base
+    ( ErrorCall )
+import Control.Monad.Except
+    ( runExceptT )
+import Control.Monad.Trans.Class
+    ( lift )
+import Test.Hspec
+    ( Spec
+    , afterAll
+    , beforeAll
+    , describe
+    , it
+    , shouldReturn
+    , shouldSatisfy
+    , shouldThrow
+    )
+
+import qualified Cardano.NetworkLayer.HttpBridge as HttpBridge
+
+port :: Int
+port = 1337
+
+spec :: Spec
+spec = do
```

Comment:
Do we test (or it is reasonable to test) scenario when the tip refers to boundary block (ie., not main, but genesis). When talking with Byron nodes, one should take that into consideration. The consumers were aware of both types of blocks, handled two types of header hashes. But in blocks were dealt differently  two places of the code, in chain block (if I remember well) all blocks were downloaded. Block coming from using listener mechanism were filtered upstream and the consumer (eg. wallet) were delivered only main blocks

Reply:
We do not because this is handled by our network layer. We do test fetching a whole epoch (so, when specifying a slot start at `slot_number = 0`). So, in contrary with the legacy wallet, we won't have to handle epoch boundaries blocks in the wallet logic (at this stage at least), since they are filtered out by the network layer. 
When dealing with Haskell Shelley node, we'll have to reconsider some of that and will most probably have to apply some of the ledger rules to the EBB in order to keep track of a few protocol parameters.